### ISSUE TITLE
General questions about handling NSFW content

### ISSUE THREAD

**Issue by kizyahaha** [kizyahaha/MSB_website_design#145]
1.)  Should users be able to see their own posts regardless of NSFW preferences?
- This adds special use cases which suck
- Billy's opinion is that if they have a "don't show" setting, that we shouldn't allow them to post NSFW content.  But this seems odd to me.  What if I just want to venture forth from time to time?

2.)  Should we show people that a rant is NSFW even if they have the "show me" preference?
- does this then become stupidly close to the "warn me" preference?
- We could just add a NSFW to the rant preview, but it's already crowded.

3.)  What if someone has "don't show" or "warn me" but then they visit an entirely NSFW subbox?
- I would lean toward just displaying a single warning modal and then if they proceed just display as if they had "show me all"
- Bill would say don't let them in if they have "don't show" and "warn me" should act exactly the same.

**Comment by kizyahaha** [comment]
1.) If users have "don't show" we should treat their own NSFW rants as "warn me"

2.) Yes, we need to show people that it's NSFW even if they have the "show me" preference.  Just a little tag in the rant preview should suffice.

3.) Figure out if we're even having subboxes first.

**Comment by kizyahaha** [comment]
I'm not so sure anymore about #1.  Now I'm leaning toward no special case if isOwner.  My logic is as follows...
- What if I am at work or church or whatever and I don't want to see ANY nsfw content?  But when I'm at home, I like to post pictures of Brazilian fart porn.  That is my right.  It's not such an extreme example.  Or what if I'm showing my family some funny post I made?  I don't want them seeing the nsfw post I made 5 minutes earlier where I display my weiner.  I'd want my "don't show" preference to be respected, not bent or broken just because I posted the nasty.
- If we don't have the special case, people still can see their own nsfw content.  There is a simple and logical solution - just change your preferences.  But if we DO have the special case, there is NO work around.  You cannot filter ALL nsfw content (unless we add even more preference options, but that could get confusing).  Between the two of these, I would want to go with the option that doesn't close any doors.
- It's just plain neater to not have the special case which will undoubtedly result in a thousand if-else statements.
